src/main/java/com/liferay/faces/util/jsp/internal/PageContextStringImpl.java

```diff
@@ -314,6 +314,11 @@ public void initialize(Servlet servlet, ServletRequest request, ServletResponse
 	}
 
 	@Override
+	public BodyContent pushBody() {
+		return new BodyContentStringImpl(new JspWriterStringImpl());
```

Comment:
@ngriffin7a, [this method returns `null` by default](http://grepcode.com/file/repo1.maven.org/maven2/javax.servlet/servlet-api/2.3/javax/servlet/jsp/PageContext.java#PageContext.pushBody%28%29). So my changes shouldn't break anything. I also tested `portal:navbarSearch` and `portal:inputRichText` and they worked correctly.